source/MaterialXCore/Element.cpp

```diff
@@ -476,6 +476,31 @@ TypeDefPtr TypedElement::getTypeDef() const
     return resolveNameReference<TypeDef>(getType());
 }
 
+bool TypedElement::validate(string* message) const
+{
+    bool res = true;
+
+    if (hasType())
+    {
+        // check that the type specified is declared as a type in the document
+        const auto& typeStr = getType();
+        bool isValidType = getDocument()->getTypeDef(typeStr) != nullptr;
+
+        if (!isValidType) {
+            // there is one special type "multioutput" that can be used on node
```

Comment:
As I recall there is at least one one which has no type. I think it's the `background` or `dot` or some "application" node(s). It's special cased elsewhere but (sorry) I don't recalled where.

Reply:
Thanks for flagging this, obviously if we decide to move forward with this, then I'll make sure we're passing all the unit tests, and hopefully that flags any existing corner cases as you mention.